tests/test_concurrency.py

```diff
@@ -9,6 +9,10 @@
     sys.version_info < (3, 7),
     reason="Requires Python 3.7+ for AbstractEventLoop.start_tls()",
 )
+#  FIX: Why doesn't return result
+@pytest.mark.xfail(
+    sys.platform == "win32", reason="StreamReader.read doesn't return result on Win32"
+)
```

Comment:
A few questions:

- What is happening exactly? Is `.read()` just hanging forever? We should try to make the reason as explicit as possible.
- Is this affecting all Python versions, or e.g. just 3.8?

Also, I don't think `xfail` supports conditions, so we should move this inside the test

```python
async def test_start_tls_on_socket_stream(https_server):
    if sys.platform == "win32":
        pytest.xfail("StreamReader.read doesn't return result on win32")
```

Reply:
the connection is dropped and doesn't raise an error. return data is `b''`. I added these as comment and removed `xfail` for entire test.

Reply:
> Is this affecting all Python versions, or e.g. just 3.8?

I will test this

Reply:
> the connection is dropped and doesn't raise an error. return data is b''.

Actually, that sounds like the bug that's affecting us and making the build fail on 3.8 currently: https://github.com/encode/httpx/pull/423#discussion_r330871311

Reply:
btw I removed 3.8 on Windows in other PR #457

Reply:
> > the connection is dropped and doesn't raise an error. return data is b''.
> 
> Actually, that sounds like the bug that's affecting us and making the build fail on 3.8 currently: [#423 (comment)](https://github.com/encode/httpx/pull/423#discussion_r330871311)

I tested this on 3.7 on Windows.